### ISSUE TITLE
[DISTANCE] LOW:MapBox doesn't work on AdHoc builds 

### ISSUE THREAD

**Issue by thienlnam** [Expensify/App#31665]
### Problem

A C+ team member [reported](https://expensify.slack.com/archives/C01GTK53T8Q/p1696842071012559) that MapBox isn't working on AdHoc builds. 

![Screenshot 2023-10-09 at 11 59 36 AM](https://github.com/Expensify/Expensify/assets/10736861/6e7c0722-0520-4915-86c2-bff9b0d93f4d)

### Solution

Figure out why the token isn't working in the AdHoc builds

Some additional context from @roryabraham :

>I think in the case of AdHoc builds, the MAPBOX_SDK_DOWNLOAD_TOKEN is available since the action is manually triggered by an Expensify employee and then just checks out the branch in the fork.

>So it seems there's some other reason this isn't working

>Example build where it seems setup-mapbox-sdk.sh runs as expected on a test build for a PR from a fork: [Expensify/App/actions/runs/6692885376/job/18183021632](https://github.com/Expensify/App/actions/runs/6692885376/job/18183021632)

**Comment by thienlnam** [comment]
Thanks, updated - affects mobile devices that use the token so native iOS and Android

**Comment by Tony-MK** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
MapBox doesn't work on AdHoc builds

### What is the root cause of that problem?

![Console Log](https://github.com/Expensify/App/assets/22982173/96f91dc7-a180-4346-bf5b-930aa6cfca47)
With AdHoc builds there are strict content security policies such as `script-src 'self'`. The mapbox by default creates a URL from a blob and sets it as the worker script src. This violates that policy and that is why we see the console log above in dev tools.



https://github.com/visgl/react-map-gl/blob/e11b41eefe73ff773bbec1db73277e4aca1f1bef/src/components/map.tsx#L76-L79

The previous error affects the `mapbox.supported` function to always return false in this situation because the mapbox script worker URL can not to created from a blob. That is why see the console log below.

![Console Log 1](https://github.com/Expensify/App/assets/22982173/3096fb14-a268-49b8-af3e-e74158fee636)

I got the main idea from a similar [discussion](https://github.com/visgl/react-map-gl/discussions/2027#discussion-4512614) in the [react-map-gl](https://github.com/visgl/react-map-gl) repo. The maintainers update the code above to `setGlobals` before checking if is supported because setting `workerUrl` did not work.


### What changes do you think we should make in order to solve the problem?


According to the discussion mentioned previously, it suggests adding the map prop to `workerUrl` to a static asset library. Such as: 
```
workerUrl=`https://api.mapbox.com/mapbox-gl-js/v${mapboxgl.version}/mapbox-gl-csp-worker.js`
```

This works because the worker will be forced to fetch the worker thus adhering to the `script-src 'self'` policy. However, we should probably add a condition to check if such policies exist so we don't interfere with other types of builds.

diffs or write walls of text. Do not create PRs unless you have been hired for this job.

**Comment by dylanexpensify** [comment]
bump @fedirjh on above proposal!

**Comment by Tony-MK** [comment]
@fedirjh I have failed to reproduce the bug on this Android ad-hoc build available here https://ad-hoc-expensify-cash.s3.amazonaws.com/android/30968/index.html. The map worked perfectly fine on a physical device. Could you please try to reproduce the bug because it seems fixed? 

However, the web application still refuses to load the map as before.
![Screenshot 2023-12-05 at 09 47 24](https://github.com/Expensify/App/assets/22982173/600f90fb-cef3-411a-a6f6-dc0a095f2a1a)

**Comment by thienlnam** [comment]
Yup that sounds good to me, we can just test by generating the ad hoc on the PR and seeing if it works on web. It's currently the only proposal we have so far that seems to address this so we'll just move forward with that

**Comment by thienlnam** [comment]
Thanks for investigating @fedirjh. I'll look into that for the ad hoc build

**Comment by thienlnam** [comment]
After the CSP changes, this still seems to be failing 
<img width="587" alt="Screenshot 2024-01-08 at 3 27 55 PM" src="https://github.com/Expensify/App/assets/30609178/39d73804-7ce8-4951-a4c4-f08c66d90705">

I wonder if we need both the CSP and front-end changes. Going to try generating another adhoc on the previous PR

**Comment by thienlnam** [comment]
Ah I see - yeah I can add those few urls that are missing from connect-src

**Comment by thienlnam** [comment]
<img width="374" alt="Screenshot 2024-01-10 at 10 39 03 AM" src="https://github.com/Expensify/App/assets/30609178/05cc517c-0b9c-4ec9-a6b8-f4bacad470fb">

We're in business 🎉 

There are a few other console errors I can clean up later but for the most part - this works now
<img width="426" alt="Screenshot 2024-01-10 at 10 39 17 AM" src="https://github.com/Expensify/App/assets/30609178/54b14f55-2d8f-4c2c-a9f8-0c7f2c1696df">

**Comment by thienlnam** [comment]
Ah yeah looks like it is missing the / in the URL, the api.mapbox.com one I'm not totally sure, we do have them in the connect-src now

**Comment by thienlnam** [comment]
We're done here! Working as expected with the latest adhoc builds. 

The solution we ended up needing to go with was the BE CSP change. This wasn't the initial route we went down, but after exploring the FE fix @fedirjh and @Tony-MK discovered that we would need to do a CSP change. However, it was an alternate solution listed in the initial proposal. 

I'm not sure how we should handle payments in this case - I definitely think we should do some sort of compensation for both for their time despite it turning into a BE fix. 

cc @dylanexpensify

**Comment by thienlnam** [comment]
cc @dylanexpensify Have we handled payment here yet?

**Comment by dylanexpensify** [comment]
Payment summary:
- Contributor: @fedirjh $125
- Contributor+: @Tony-MK $125

**Comment by dylanexpensify** [comment]
sent new offers to both!

**Comment by Tony-MK** [comment]
Thanks @dylanexpensify. Accepted.

**Comment by dylanexpensify** [comment]
done!

**Comment by Tony-MK** [comment]
Hey @dylanexpensify,

Could you help cancel the initial [offer](https://www.upwork.com/nx/wm/offer/28014257)?

Thanks.

**Comment by dylanexpensify** [comment]
Done! cc @Tony-MK
